.github/workflows/integration_tests.yml

```diff
@@ -16,6 +16,12 @@ on:
 permissions:
   contents: read
 
+concurrency:
+  # Don't run this workflow concurrently on the same branch
+  group: ${{ github.workflow }}-${{ github.ref }}
+  # For PRs, don't wait for completion of existing runs, cancel them instead
+  cancel-in-progress: ${{ github.event_name == 'pull_request' }}
```

Comment:
Just a heads up, I'm not sure this'll quite work because this workflow doesn't have a pull request trigger. It triggers on pushes to all branches, so maybe what you want is to cancel-in-progress if the branch isn't your main branch. https://docs.github.com/en/actions/learn-github-actions/contexts#github-context has docs on what all is available to you in the expressions here